### ISSUE TITLE
Restore fails at post-data stage

### ISSUE THREAD

**Issue by janmeier** [GreenmaskIO/greenmask#75]
About half of the time when we run `greenmask restore`, the `post-data` stage fails with the following error:

```
FTL ../home/runner/work/greenmask/greenmask/cmd/greenmask/cmd/restore/restore.go:68 > fatal error="post-data stage restoration error: cannot start transaction: write failed: write tcp 192.168.1.212:56750->192.168.3.203:5432: write: connection reset by peer" pid=354151
```

My guess is that since the same connection is being reused in `restore.Run` https://github.com/GreenmaskIO/greenmask/blob/c21cc3b99fbfd61d842007658337d466c65d6bca/internal/db/postgres/cmd/restore.go#L480C19-L480C22, and since the data restoring stage takes several hours, the connection is timed out by the server.

I can try my hand at creating a PR that opens a separate connection for each stage, if you want?

**Comment by wwoytenko** [comment]
Hi! Thank you for reporting. 

Sure, feel free to make any contribution and open PR.

**Comment by wwoytenko** [comment]
Could you check in the log what kind of timeout terminates this connection?
`idle_in_transaction_session_timeout`
`idle_session_timeout`
`statement_timeout`

**Comment by wwoytenko** [comment]
@janmeier 
I've just checked carefully the case you described. You're right, we could open a new connection only when it is required avoiding keeping it idle. Feel free to ask me if you need any assistance or help with implementation.

**Comment by wwoytenko** [comment]
@janmeier 

I've received the same issue from another user. If you don't mind and aren't going to contribute, I can fix it and publish it in next week's release

**Comment by wwoytenko** [comment]
This resolved. It will be published in v0.1.11

**Comment by wwoytenko** [comment]
Published in v0.1.11

**Comment by janmeier** [comment]
> @janmeier
> 
> I've received the same issue from another user. If you don't mind and aren't going to contribute, I can fix it and publish it in next week's release

Thanks a lot! I mostly offered to contribute, since I felt like I couldn't just keep opening issues demanding new features and bug fixes. But I haven't found the time yet, so I'm very happy you did!
